statsmodels/multivariate/factor.py

```diff
@@ -0,0 +1,272 @@
+# -*- coding: utf-8 -*-
+
+from statsmodels.base.model import Model
+from .factor_rotation import rotate_factors, promax
+
+try:
+    import matplotlib.pyplot as plt
+    missing_matplotlib = False
+except ImportError:
+    missing_matplotlib = True
+
+if not missing_matplotlib:
+    from .plots import plot_scree, plot_loadings
+
+import numpy as np
+from numpy.linalg import eigh, inv, norm, matrix_rank
+import pandas as pd
+
+from statsmodels.iolib import summary2
+
+
+class Factor(Model):
+    """
+    Factor analysis
+
+    .. [1] Hofacker, C. (2004). Exploratory Factor Analysis, Mathematical Marketing.
+    http://www.openaccesstexts.org/pdf/Quant_Chapter_11_efa.pdf
+
+    Supported rotations:
+        'varimax', 'quartimax', 'biquartimax', 'equamax', 'oblimin',
+        'parsimax', 'parsimony', 'biquartimin', 'promax'
+
+    Parameters
+    ----------
+    endog : array-like
+        Variables in columns, observations in rows
+
+    n_factor : int
+        The number of factors to extract
+
+    method : str
+        Specify the method to extract factors.
+        'pa' - Principal axis factor analysis
+
+    """
+    def __init__(self, endog, n_factor, exog=None, method='pa', **kwargs):
+        if n_factor <= 0:
+            raise ValueError('n_factor must be larger than 0! %d < 0' %
+                             (n_factor))
+        if n_factor > endog.shape[1]:
+            raise ValueError('n_factor must be smaller or equal to the number'
+                             ' of columns of endog! %d > %d' %
+                             (n_factor, endog.shape[1]))
+        self.n_factor = n_factor
+        self.loadings = None
+        self.communality = None
+        self.eigenvals = None
+        self.method = method
+        super(Factor, self).__init__(endog, exog)
+
+    def fit(self, **kwargs):
```

Comment:
use explicit keywords because they are part of the documentation

naming convention maxiter=100, tol=1e-8  (copied from GLM)